core/src/main/java/jenkins/util/io/PathRemover.java

```diff
@@ -113,6 +115,14 @@ public void forceRemoveRecursive(@NonNull Path path) throws IOException {
     public interface RetryStrategy {
         boolean shouldRetry(int retriesAttempted);
 
+        /**
+         * Whether the current deletion attempt should be aborted early based on the errors seen so far.
+         * Additional attempts may still occur according to {@link #shouldRetry} even if the current attempt is aborted.
+         */
+        default boolean shouldAbort(List<IOException> accumulatedErrors) {
```

Comment:
I was not sure on the best way to integrate this configuration into `PathRemover`, so for now I just added it as a method in `RetryStrategy`. I could also see a case for creating a distinct `AbortStrategy` field with just this method, or just adding a `maxErrorsBeforeAbort` field directly in `PathRemover`, but that seemed like it didn't really fit the existing design.